### ISSUE TITLE
Deal with case insensitive file systems.

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/git-subrepo#157]
See #156

I think the right thing to do here is autocorrect `foo/bar` into `Foo/Bar` for
subrepos called `Foo/Bar` that are referenced with `foo/bar` on the command
line.

I can't seem to figure out how to get Bash to tell me the real name of
`foo/bar` is `Foo/Bar`.

**Comment by grimmySwe** [comment]
What about this? then we at least get an error if its not in the actual repo.
```
-  if [[ ! -f $gitrepo ]]; then
+  o "Test that the file exists in git"
+  FAIL=false RUN git cat-file -e HEAD:$subdir
+  if [[ ! OK || ! -f $gitrepo ]]; then
     error "'$subdir' is not a subrepo."
   fi

```

**Comment by ingydotnet** [comment]
@grimmySwe I thought to do something like that at first. But the nicer thing to do is succeed rather than fail. Especially since there cannot be any real ambiguity.

**Comment by grimmySwe** [comment]
I agree, but isn't this a bigger problem, cause git will actually care for cases or?
http://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git

**Comment by ingydotnet** [comment]
All I'm suggesting is that when someone enters a subrepo of `foo` when they meant `Foo`, that we treat it as if they entered `Foo`.

Only problem is that haven't figured out how to do that in Bash yet.

**Comment by robe070** [comment]
I read the link that @Grimmyswe provided and it talks about git performing
that mapping of foo to Foo. Probably doesn't help you though.

warm regards

Rob Goodridge

**Comment by grimmySwe** [comment]
@ingydotnet What do you think about adding the quick fix now to make sure that we avoid problems like #156 
Then we have time to figure out a better solution like the one you propose, where you actually ignore case somehow.

**Comment by ingydotnet** [comment]
I think my fix is trivial. Just need to locate the command.

**Comment by ingydotnet** [comment]
This looks like it will probably work:

```
find . -type d | grep -i "^$subdir\$"
```

**Comment by grimmySwe** [comment]
@ingydotnet What is the status of this?
